module/src/components/stepper/stepper.component.tsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable no-nested-ternary */
+
+import * as React from 'react';
+
+import { IconSet, IIcon } from '../..';
+import { Icon } from '../icon';
+import { IStatusProps, Status } from '../status';
+
+export interface IStepperStep extends Pick<IStatusProps, 'pending' | 'error'> {
+  /** (IIcon) the icon for this step */
+  icon?: IIcon<IconSet>;
+
+  /** (number) the index of this step from 0 */
+  index?: number;
+
+  /** (string) the name of this step */
+  name?: string;
+
+  /** (boolean) is this step disabled - stops the onChange from firing */
+  disabled?: boolean;
+
+  /** (boolean) is this step complete */
+  isComplete?: boolean;
+}
+
+export interface IStepperStepProps extends IStepperStep, Pick<IStatusProps, 'spinnerIcon' | 'errorIcon'> {
+  /** (boolean) is this step the current step */
+  isCurrent?: boolean;
+
+  /** (boolean) is this step previous to the current step */
+  isPrevious?: boolean;
+
+  /** ((event) => void) fired when the user clicks on the stepper step */
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+
+  /** (IIcon) the icon to render if isComplete is set to true - leave blank to not change */
+  completeIcon?: IIcon<IconSet>;
+
+  /** (boolean) is small (no icon or index) */
+  small?: boolean;
+
+  /** (boolean) is the next one small (no icon or index) used to position the line between them */
+  nextIsSmall?: boolean;
+}
+
+export const StepperStep = React.forwardRef<HTMLButtonElement, IStepperStepProps>(
+  (
+    {
+      isCurrent,
+      isComplete,
+      isPrevious,
+      onClick,
+      icon,
+      children,
+      name,
+      disabled,
+      index,
+      completeIcon,
+      small,
+      nextIsSmall,
+      spinnerIcon,
+      error,
+      errorIcon,
+      pending,
+    },
+    ref
+  ) => {
+    const onClickEvent = React.useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        if (!disabled) {
+          onClick?.(event);
+        }
+      },
+      [onClick, disabled]
+    );
+
+    return (
+      <button
+        className="arm-stepper-step"
+        aria-current={isCurrent && 'step'}
+        onClick={onClickEvent}
+        data-current={isCurrent}
+        data-previous={isPrevious}
+        data-complete={isComplete}
+        data-pending={pending}
+        data-error={error}
+        data-small={small}
+        data-next-small={nextIsSmall}
+        disabled={disabled}
+        title={name}
+        ref={ref}
+      >
+        <div className="arm-stepper-step-icon">
+          {error || pending ? (
+            <Status error={error} errorIcon={errorIcon} pending={pending} spinnerIcon={spinnerIcon} />
+          ) : completeIcon && isComplete ? (
+            <Icon iconSet={completeIcon.iconSet} icon={completeIcon.icon} />
+          ) : icon ? (
+            <Icon iconSet={icon.iconSet} icon={icon.icon} />
+          ) : (
+            typeof index !== 'undefined' && <p className="arm-stepper-step-icon-index">{index! + 1}</p>
```

Comment:
A minor improvement, but using a number check here rather than a type check will mean you can avoid the `!` when you use it as a number (I think):
```
Number.isInteger(index) && <p className="arm-stepper-step-icon-index">{index + 1}</p>
```

Reply:
Hmm strict mode is still being a lil jerk after this change, so the ! has to stay